### DISCUSSION TITLE
Sharepoint files

### DISCUSSION THREAD

**Discussion by mmc00** [ropensci/targets#513] [Help] [answered]
Hello!

I would like to now how can I handle if a Sharepoint URL changes. For example, I have an private url from Sharepoint that I can download throught the [Microsoft365R](https://github.com/Azure/Microsoft365R) package. My current plan is something like this:

list(
  tar_target(
    obj_download,
    "https://sharepoint.com/***",
    format = "url"
  )
)

But the obj_download fails, because it tries to get it but needs the credentials from Microsoft365R. If I keep it just as string, I don't keep the track if the file has change.

**Comment by wlandau** [comment, marked as answer]
An alternative to the URL format would be something like 

```r
tar_target(
  obj_download,
  download_file(),
  cue = tarchetypes::tar_cue_force(condition = etag_file())
)
```

where `download_file()` is a custom function to download the data and `etag_file()` is a custom function to get the ETag. Both would use `Microsoft365R`.
